src/components/ContextualSaveBar/ContextualSaveBar.tsx

```diff
@@ -1,32 +1,36 @@
 import * as React from 'react';
+import compose from '@shopify/react-compose';
 import isEqual from 'lodash/isEqual';
-import {
-  ContextualSaveBarProps,
-  FrameContext,
-  frameContextTypes,
-} from '../Frame';
+import {ContextualSaveBarProps, FrameContext, Consumer} from '../Frame';
+import withContext from '../WithContext';
+import {WithContextTypes} from '../../types';
+import {withAppProvider, WithAppProviderProps} from '../AppProvider';
 
 // The script in the styleguide that generates the Props Explorer data expects
 // a component's props to be found in the Props interface. This silly workaround
 // ensures that the Props Explorer table is generated correctly, instead of
 // crashing if we write `ContextualSaveBar extends React.Component<ContextualSaveBarProps>`
 interface Props extends ContextualSaveBarProps {}
+export type ComposedProps = Props &
+  WithAppProviderProps &
+  WithContextTypes<FrameContext>;
 
-class ContextualSaveBar extends React.PureComponent<Props, never> {
-  static contextTypes = frameContextTypes;
-  context: FrameContext;
-
+class ContextualSaveBar extends React.PureComponent<ComposedProps, never> {
   componentDidMount() {
-    this.context.frame.setContextualSaveBar(this.props);
+    const {props} = this;
+    const {polaris, context, ...rest} = props;
+    props.context.frame.setContextualSaveBar(rest);
```

Comment:
It feels a bit gross having all of this available on a `context` object inside of `props`. Do we think there might be a cleaner way to organize this?

https://github.com/Shopify/polaris-react/blob/master/src/types.ts#L243

Reply:
you wouldn't need the `props.` since you are pulling off the context.

Reply:
I think this is just the reality because `setContextualSaveBar` accepts `ContextualSaveBarProps` which never included `polaris`.

Reply:
In this case I was more referring to the HOC that adds all of our context to a `context` object, as shown in the linked code in the original comment. Wondering if this is the best pattern going forward.

cc: @AndrewMusgrave

Reply:
The other option is it spread context onto the components props rather than placing context on props.

E.g.
```tsx
this.props.context.a
// or
this.props.a
```

However if we don't namespace the context we could encounter naming collisions.

Reply:
I think that keeping it namespaced to `context` makes sense here. I'm good with this as a pattern.